### PR TITLE
tsserver et al cannot be executed #95

### PR DESCRIPTION
**Pull request by stephan-herrmann**
Scripts for `.bin` as created by `npm` under windows + git bash, after disabling other language servers which cannot be installed on windows.

### CODE REVIEW

File: org.eclipse.wildwebdeveloper/language-servers/.gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /node_modules/
+!/node_modules/.bin
```
Comment (mickaelistria) [inline org.eclipse.wildwebdeveloper/language-servers/.gitignore]: I don't think we should put binaries on SCM herel it's a bad practice and is adding 1 extra set of artifacts to think about maintaining. Those should be provisioned at build time instead.

File: org.eclipse.wildwebdeveloper/pom.xml [hunk 2]
```diff
@@ -55,6 +66,7 @@
 							<executable>npm</executable>
 							<arguments>
 								<arg>install</arg>
+								<arg>--no-bin-links</arg>
```
Comment (mickaelistria) [inline org.eclipse.wildwebdeveloper/pom.xml]: That's interesting. Maybe there is an option to copy the binaries instead of linking them?

**Comment by stephan-herrmann** [comment]
Giving up on this for now, sorry.

Results: in a previously deployed instance of wildwebdeveloper, several alternatives worked fine:

- repaired symlink
- node script as shown in the patch in the issue
- shell script as generated by npm install on windows (the one that checks if on CYGWIN)

During surefire tests I could get none of these to work. Special difficulty: while I know that cli.js was found & executed I didn't succeed to make any visible logging from js to figure out where exactly it breaks.

Interestingly, npm install on windows does create scripts instead of symlinks, but that build is broken for any of the vscode language servers :(

File: org.eclipse.wildwebdeveloper/language-servers/.gitignore (hunk 1 above)
Comment (stephan-herrmann) [reply org.eclipse.wildwebdeveloper/language-servers/.gitignore]: > I don't think we should put binaries on SCM herel it's a bad practice and is adding 1 extra set of artifacts to think about maintaining. Those should be provisioned at build time instead.

Feel free to change, but I'm unable to see a simpler alternative (and don't know enough about npm to make it behave "a bit like windows" (i.e., create scripts not links) while running on linux).
I also wouldn't consider scripts as binaries :)

Comment (mickaelistria) [reply org.eclipse.wildwebdeveloper/language-servers/.gitignore]: Right, I used the wrong terms. It's more that I'm reluctant to check-in 3rd party code in general.

**Comment by mickaelistria** [comment]
Further investigation show that best place to improve this is the language server, and that an improvement is already implemented. Replaced this PR by https://github.com/eclipse/wildwebdeveloper/pull/99

**Pull request closed without merging**